examples/main/README.md

```diff
@@ -272,7 +272,7 @@ These options help improve the performance and memory usage of the LLaMA models.
 
 ### Prompt Caching
 
--   `--prompt-cache FNAME`: Specify a file to cache the model state after the initial prompt. This can significantly speed up the startup time when you're using longer prompts. The file is created during the first run and is reused and updated in subsequent runs.
+-   `--prompt-cache FNAME`: Specify a file to cache the model state after the initial prompt. This can significantly speed up the startup time when you're using longer prompts. The file is created during the first run and is reused and updated in subsequent runs. **Note**: Restoring a cached prompt does not imply restoring the exact state of the session at the point it was saved. So even when specifying a specific seed, you are not guaranteed to get the same sequence of tokens as the original generation.
```

Comment:
I thought about that, but we'd also have to add that this _only_ applies with the same generation settings and same exact build. Anything that changes how sampling occurs or even causes one less or one more random number to be generated at some point will throw the whole thing out of whack.

Also I think some BLAS implementations also can cause output to be non-deterministic even when meeting the other requirements (at least whenever I run the main example with cuBLAS I get a warning about when specifying a seed).

So it kind of seemed like it would just confuse the user to explain all that, and the alternative is to not tell them the whole truth. Maybe one way to deal with it would be just phrase it in a vague way like "generally speaking, if you use same seed and prompt you'll get the same result when restoring a session but changing settings or using a different version can affect this". 

What do you think?

Reply:
Yeah, I see what you're saying; I forgot that some BLAS implementations can cause things to diverge.

The docs don't mention it anywhere at this point, not even under `--seed`. I think perhaps it's a caveat that's just understood by those BLAS users?

It might be worth adding a section on BLAS to the README that explains these caveats. (Not in this patch.)

Reply:
I do believe I can edit the PR, but I also think you can just click accept and it'll merge the changes. That said, if f you're just trying to keep your hands clean, I'll approve it as is. 😄

Reply:
> That said, if f you're just trying to keep your hands clean

I guess you could call it that. :) I wouldn't really add that myself because it's (at least technically) incorrect. Like I said though, I'm okay with _you_ committing it/editing this pull/whatever.

The main thing I care about is the changes to the actual code.